### ISSUE TITLE
`std::filesystem::path` doesn't exist on macOS < 10.15

### ISSUE THREAD

**Issue by quentinmit** [fredemmott/StreamDeck-CPPSDK#9]
Stream Deck supports [macOS 10.11+](https://help.elgato.com/hc/en-us/articles/360027959372-Elgato-Stream-Deck-Technical-Specifications) but the change in 3.0.0 to use `std::filesystem` means that it's impossible to compile a plugin for macOS < 10.15 (You get the not-helpful `'path' is unavailable: introduced in macOS 10.15` compiler error).

**Comment by fredemmott** [comment]
I missed this as it builds fine when 'use latest SDK, but limit to 10.14 compatibility' is specified, both locally and on github actions.

Use v2.x instead if you need compatiblity with older MacOS.

**Comment by fredemmott** [comment]
Thanks - I've also updated the release announcement.

**Comment by fredemmott** [comment]
A workaround for https://github.com/chriskohlhoff/asio/issues/1090 is also needed - though surprisingly this is just a warning rather than an error:

```
/Users/fred/code/StreamDeck-AudioSwitcher/build/_deps/streamdecksdk-build/Vendor/asio_source-prefix/src/asio_source/asio/include/asio/detail/memory.hpp:88:15: warning: 'aligned_alloc' is only available on macOS 10.15 or newer [-Wunguarded-availability-new]
```

**Comment by fredemmott** [comment]
This does the trick in the plugin, going to try adding it here:

```
+if (APPLE AND "${CMAKE_OSX_DEPLOYMENT_TARGET}" VERSION_LESS "10.15")
+  add_compile_definitions("ASIO_DISABLE_STD_ALIGNED_ALLOC=1")
+endif ()
```
